### PR TITLE
Fixes #37251: Error occuring for remote browser

### PR DESCRIPTION
**Pull request by dosas**
Selenium::WebDriver::Error::WebDriverError: You are trying to work with something that isn't a file.

### CODE REVIEW

File: test/integration/foreman_puppet/smartclass_parameter_js_test.rb [hunk 1]
```diff
@@ -74,6 +74,7 @@ class SmartclassParameterJSTest < IntegrationTestWithJavascript
       within(:xpath, '//table') do
         click_link 'ssl'
       end
+      page.driver.browser.file_detector = nil if page.driver.browser.respond_to?(:file_detector=)
```
Comment (stejskalleos) [inline test/integration/foreman_puppet/smartclass_parameter_js_test.rb:77]:
```suggestion
      page.driver.browser&.file_detector = nil
```

Comment (stejskalleos) [inline test/integration/foreman_puppet/smartclass_parameter_js_test.rb:77]: If this is a workaround for buggy behavior, I would suggest adding a comment with a link to the Redmine or a bug report.
Also, shouldn't this be a part of before/after test setup actions?

**Review by stejskalleos — CHANGES_REQUESTED** [review]

File: test/integration/foreman_puppet/smartclass_parameter_js_test.rb (hunk 1 above)
Comment (dosas) [reply test/integration/foreman_puppet/smartclass_parameter_js_test.rb:77]: AFAIK this is a problem with the selenium ruby bindings https://stackoverflow.com/questions/70441796/selenium-webdriver-for-aws-device-farm-error-when-sending-period-keystroke-t

another more implicit fix is described here https://github.com/theforeman/foreman_puppet/issues/382 to just rename `test` to something that is not locally available

File: test/integration/foreman_puppet/smartclass_parameter_js_test.rb [hunk 2]
```diff
@@ -74,6 +74,7 @@ class SmartclassParameterJSTest < IntegrationTestWithJavascript
       within(:xpath, '//table') do
         click_link 'ssl'
       end
+      page.driver.browser&.file_detector = nil
```
Comment (stejskalleos) [inline test/integration/foreman_puppet/smartclass_parameter_js_test.rb:77]: Can you add a comment  to the code as well, with comment link to the Redmine? So it's immediately clear what's the reasoning behind this workaround.

Comment (dosas) [reply test/integration/foreman_puppet/smartclass_parameter_js_test.rb:78]: done

File: test/integration/foreman_puppet/smartclass_parameter_js_test.rb [hunk 3]
```diff
@@ -74,6 +74,8 @@ class SmartclassParameterJSTest < IntegrationTestWithJavascript
       within(:xpath, '//table') do
         click_link 'ssl'
       end
+      # see: https://projects.theforeman.org/issues/37251
+      page.driver.browser&.file_detector = nil
```
Comment (stejskalleos) [inline test/integration/foreman_puppet/smartclass_parameter_js_test.rb:78]: Hmm, tests are failing now on:
```
NoMethodError: undefined method `file_detector=' for #<Selenium::WebDriver::Chrome::Driver:0x..fda95f2e33cc25c50 browser=:chrome_headless_shell>
    foreman_puppet/test/integration/foreman_puppet/smartclass_parameter_js_test.rb:78:in `block in <class:SmartclassParameterJSTest>'
rake aborted!
```

But why? There is the safe guard ...

Comment (dosas) [reply test/integration/foreman_puppet/smartclass_parameter_js_test.rb:78]: I don't know. I would propose to switch back to the version that worked.

Comment (stejskalleos) [reply test/integration/foreman_puppet/smartclass_parameter_js_test.rb:78]: Yop, agree ... Sorry for the not working suggestion

Comment (dosas) [reply test/integration/foreman_puppet/smartclass_parameter_js_test.rb]: Nevermind. I thought it would work too. I updated the PR.

**Comment by stejskalleos** [comment]
Thanks @dosas

**Pull request merged**